library/ssl_cli.c

```diff
@@ -1277,6 +1377,14 @@ static int ssl_write_client_hello( mbedtls_ssl_context *ssl )
     ext_len += olen;
 #endif
 
+#if defined(MBEDTLS_SSL_DTLS_SRTP)
+    if( ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM )
+    {
+        ssl_write_use_srtp_ext( ssl, p + 2 + ext_len, &olen );
```

Comment:
In line with the pattern for other extension writing functions, could you pass the `end` pointer as the last parameter and include appropriate bounds checks in `ssl_write_use_srtp_ext`?

Reply:
NB: The code doesn't seem to be uniform in that regard: Client-side extension writing functions accept a end-of-buffer parameter, while server-side extension writing functions don't.